sample-plugin/src/main/java/com/github/jackchen/plugin/sample/instrumentation/SampleAsmClassVisitorFactory.kt

```diff
@@ -3,24 +3,52 @@ package com.github.jackchen.plugin.sample.instrumentation
 import com.android.build.api.instrumentation.AsmClassVisitorFactory
 import com.android.build.api.instrumentation.ClassContext
 import com.android.build.api.instrumentation.ClassData
-import com.android.build.api.instrumentation.InstrumentationContext
 import com.android.build.api.instrumentation.InstrumentationParameters
+import com.github.jackchen.android.sample.api.Extension
+import com.github.jackchen.android.sample.api.Register
+import com.github.jackchen.android.sample.api.TestCase
 import com.github.jackchen.plugin.sample.visitor.SampleClassVisitor
-import org.gradle.api.provider.Property
-import org.gradle.api.tasks.Input
-import org.gradle.api.tasks.Internal
-import org.gradle.api.tasks.Optional
 import org.objectweb.asm.ClassVisitor
-import java.io.File
 
+/**
+ * @author JackChen
+ */
+abstract class SampleAsmClassVisitorFactory :
+  AsmClassVisitorFactory<InstrumentationParameters.None> {
 
-abstract class SampleAsmClassVisitorFactory : AsmClassVisitorFactory<InstrumentationParameters.None> {
+  companion object {
+    private val ANNOTATION_CLASS_NAMES = hashSetOf<String>(
+      Register::class.java.name,
+      TestCase::class.java.name,
+      Extension::class.java.name,
+    )
+  }
 
-    override fun createClassVisitor(classContext: ClassContext, nextClassVisitor: ClassVisitor): ClassVisitor {
-        return SampleClassVisitor(nextClassVisitor)
-    }
+  override fun createClassVisitor(
+    classContext: ClassContext,
+    nextClassVisitor: ClassVisitor
+  ): ClassVisitor {
+    return SampleClassVisitor(nextClassVisitor)
+  }
 
-    override fun isInstrumentable(classData: ClassData): Boolean {
+  override fun isInstrumentable(classData: ClassData): Boolean {
+    val superClassName = classData.superClasses[0]
+    if (superClassName.isCompatActivityClassName() || superClassName.isFragmentActivityClassName()) {
+      return true
+    }
+    classData.classAnnotations.forEach {
+      if (ANNOTATION_CLASS_NAMES.contains(it)) {
         return true
+      }
     }
+    return false
+  }
+
+  private fun String.isCompatActivityClassName(): Boolean {
+    return this == SampleClassVisitor.ANDROIDX_COMPAT_ACTIVITY_CLASS_NAME.replace("/", ".")
+  }
+
+  private fun String.isFragmentActivityClassName(): Boolean {
```

Comment:
Is it better to use the Kotlin closure to reuse the code
For example:

```
fun inline  String.isParentClass(block:()->Boolean) = block()
```

Reply:
I think we can use this approach when there are more judgments. 

Also, the naming of this method is not very good, because there is no relationship between string and parent class.